lib/libzfs/libzfs_dataset.c

```diff
@@ -2722,7 +2745,9 @@ zfs_prop_get(zfs_handle_t *zhp, zfs_prop_t prop, char *propbuf, size_t proplen,
 		break;
 
 	case ZFS_PROP_ORIGIN:
-		str = getprop_string(zhp, prop, &source);
+		str = zhp->zfs_dmustats.dds_origin;
+		if (*str == '\0')
+			str = zfs_prop_default_string(prop);
 		if (str == NULL)
 			return (-1);
```

Comment:
I assume it is not possible for zhp->zfs_dmustats.dds_origin to be NULL?

Reply:
Fixed the ordering so we check for NULL before dereferencing.